RUFAS/input_manager.py

```diff
@@ -280,6 +289,216 @@ def _filter_input_data_by_metadata(self, input_data: Dict[str, Any],
+
+        Returns
+        -------
+        Tuple[Dict[str, Any], List[str], List[Tuple[str, Any]]]
+            A tuple containing:
+            - The updated input data with defaults added.
+            - A list of missing required property keys.
+            - A list of tuples where each tuple is a property key and its default value that was added.
+        """
+
+        missing_required_property_keys: List[str] = []
+        property_keys_with_default_values: List[Tuple[str, Any]] = []
+        for property_key, property_details in metadata_properties.items():
+            if property_key in ['type', 'description', 'default']:
+                continue
+            if property_key not in input_data:
+                if 'default' in property_details:
+                    input_data[property_key] = property_details['default']
+                    property_keys_with_default_values.append((property_key, property_details['default']))
+                else:
+                    missing_required_property_keys.append(property_key)
+                    continue
+
+            # Even if the property key is present or fulfilled with a default value,
+            # we still need to check if it's a nested object/array and add default values
+            # to missing properties in the nested object/array
+            if property_details['type'] == 'object':
+                nested_input_data = input_data[property_key]
+                nested_metadata_properties = metadata_properties[property_key]
+                input_data[property_key], nested_missing_required_property_keys, \
+                    nested_property_keys_with_default_values = \
+                    self._add_default_values_to_missing_properties(
+                        nested_input_data,
+                        nested_metadata_properties,
+                    )
+                missing_required_property_keys.extend(
+                    [f'{property_key}.{key}' for key in nested_missing_required_property_keys]
+                )
+                property_keys_with_default_values.extend(
+                    [(f'{property_key}.{key}', value) for key, value in nested_property_keys_with_default_values]
+                )
+
+            elif property_details['type'] == 'array':
+                input_data[property_key], nested_missing_required_property_keys, \
+                    nested_property_keys_with_default_values = \
+                    self._add_default_values_to_array_properties(
+                        input_data,
+                        property_key,
+                        property_details,
+                    )
+                missing_required_property_keys.extend(nested_missing_required_property_keys)
+                property_keys_with_default_values.extend(nested_property_keys_with_default_values)
+
+        return input_data, missing_required_property_keys, property_keys_with_default_values
+
+    def _add_default_values_to_array_properties(
```

Comment:
```suggestion
    def _add_default_values_to_array_inputs(
```
Same as above comment. The array properties already have defaults. It is the array inputs that need defaults added to them.